crates/contract/src/impersonate.rs

```diff
@@ -0,0 +1,146 @@
+use alloy_network::Network;
+use alloy_primitives::U256;
+use alloy_provider::{ext::AnvilApi, Provider};
+use alloy_rpc_types_eth::TransactionRequest;
+use alloy_transport::TransportError;
+use futures::try_join;
```

Comment:
hmm, looks like we can actually move this to where AnvilApi is defined, this doesnt necessarily need to added in the contract crate?